test.sh

```diff
@@ -3,33 +3,49 @@
 set -e
 
 # Stop all child processes on exit
-trap "pkill -P $$" EXIT
+trap cleanup EXIT
+
+cleanup() {
+    pkill -P $$
+    if [ -n "$tmpdir" ] && [ -d "$tmpdir" ]; then
+        rm -r "$tmpdir"
+    fi
+}
 
 export GOBIN="$PWD/bin"
+export PATH="$GOBIN:$PATH"
+tmpdir=
+
+mkdir -p "$GOBIN"
 
 if [ -z "$SIMPLEBLOB_TEST_S3_CONFIG" ]; then
     echo "* Using MinIO for S3 tests"
     export SIMPLEBLOB_TEST_S3_CONFIG="$PWD/test-minio.json"
 
-    # Check for existing minio
-    minio=$(which minio || true)
-
     # Fetch minio if not found
-    if [ -z "$minio" ]; then
-        #minioversion="github.com/minio/minio@v0.0.0-20220420232007-ddf84f8257c9"
-        #echo "+ go install $minioversion" > /dev/stderr
-        #go install "$minioversion"
+    if ! command -v minio >/dev/null; then
         source <(go env)
-        curl -v -o "$GOBIN/minio" "https://dl.min.io/server/minio/release/$GOOS-$GOARCH/minio"
-        minio=./bin/minio
-        chmod 6755 "$minio"
+        dst="$GOBIN/minio"
+        curl -v -o "$dst" "https://dl.min.io/server/minio/release/$GOOS-$GOARCH/minio"
+        chmod u+x "$dst"
     fi
 
     # Start MinIO
-    echo "* Starting $minio on port 34730"
+    echo "* Starting minio at address 127.0.0.1:34730"
     tmpdir=$(mktemp -d -t minio.XXXXXX)
-    "$minio" server --address 127.0.0.1:34730 --console-address 127.0.0.1:34731 --quiet "$tmpdir" &
-    sleep 3
+    minio server --address 127.0.0.1:34730 --console-address 127.0.0.1:34731 --quiet "$tmpdir" &
+    # Wait for minio server to be ready
+    i=0
+    while ! curl -s -I "127.0.0.1:34730/minio/health/ready" | grep '200 OK' >/dev/null; do
+        i=$((i+1))
+        if [ "$i" -ge 100 ]; then
+            # We have been waiting for server to start for 10 seconds
+            echo "Minio could not start properly"
+            curl -s -I "127.0.0.1:34730/minio/health/ready"
+            exit 1
+        fi
+        sleep .1
```

Comment:
From the macOS manpage:

> The sleep command allows and honors a non-integer number of seconds to sleep in any form acceptable by strtod(3).  This is a non-portable extension, but is also implemented in GNU sh-utils since version 2.0a (released in 2002).

I think this is ok for now, but it may not work on every platform.